### PR TITLE
i18n: Add option to set console keymap from xkb

### PR DESCRIPTION
**Pull request by greedy**

### CODE REVIEW

**Comment by vcunat** [comment]
In principle this looks good to me. I refactored it heavily and tried, which uncovered some wrong option names. I think this will be good: c0171a8d29.

**Comment by vcunat** [comment]
Oh, I'm sorry. There were no wrong option names.

File: nixos/modules/config/i18n.nix [hunk 1]
```diff
@@ -9,6 +9,23 @@ let
     locales = config.i18n.supportedLocales;
   };
 
+  xcfg = config.services.xserver;
+
+  ckbKeymap = pkgs.stdenv.mkDerivation {
```
Comment (globin) [inline nixos/modules/config/i18n.nix]: Typo, should be **x**kbKeymap

Comment (vcunat) [inline nixos/modules/config/i18n.nix]: I believe this was intentional. It's a local binding anyway.

Comment (globin) [inline nixos/modules/config/i18n.nix]: Mkay, too early :speak_no_evil: :coffee:

**Comment by greedy** [comment]
@vcunat I originally had the xkb options inserted as interpolations as well but changed it because I was worried about what would happen if they contained shell metacharacters. I suppose any valid values wouldn't have such characters in it though. Is there an overall nix philosophy on things like this?

**Comment by vcunat** [comment]
I see. I put the stuff into apostrophes, so the number of possible meta-characters is lowered. In your version, perhaps the variables aren't processed when substituted into the command &ndash; I'm not sure &ndash; this always seemed magic to me (and it isn't very simple).

In shells, it's typically really difficult to be completely correct in this respect (and I personally really dislike that), so in many places we aren't. I think even spaces in filenames would cause problems on a few places, because on some places we didn't quote at all.

**Pull request closed without merging**